### PR TITLE
Create Models 🚀

### PR DESCRIPTION
**Pull request by hsztan**
I create the models with their corresponding associations according to the [UML](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). I also add specific methods to manipulate data and query the database with the ORM. 😁

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def increment_user_posts_count
+    user.posts_counter += 1
+    user.save
+  end
```
Comment (rustam-tolipov) [inline app/models/post.rb]: - [x] Your method is good. However there are more clean and efficient way to it. 
 You can put the increment method after save. Like that:

```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def increment_post_comments_counter
+    post.comments_counter += 1
+    post.save
+  end
```
Comment (rustam-tolipov) [inline app/models/comment.rb]: - [x] The same method you can use here as well. 
Just like we have in model/post.rb

`  after_save :update_post_counter
...
`

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
```
Comment (rustam-tolipov) [inline app/models/user.rb:1]: - [x] [OPTIONAL] Kindly personalize your MIT license.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @hsztan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 3 above)
Comment (hsztan) [reply app/models/user.rb:1]: Will do 💪

File: app/models/post.rb (hunk 1 above)
Comment (hsztan) [reply app/models/post.rb]: Got it 💪

File: app/models/comment.rb (hunk 2 above)
Comment (hsztan) [reply app/models/comment.rb]: Smart ✅

File: app/models/post.rb (hunk 1 above)
Comment (peterrobert) [reply app/models/post.rb]: This change has been done well good work

File: app/models/comment.rb (hunk 2 above)
Comment (peterrobert) [reply app/models/comment.rb]: This change has been done well.

File: app/models/user.rb (hunk 3 above)
Comment (peterrobert) [reply app/models/user.rb:1]: Great work

**Review by peterrobert — APPROVED** [review]
Hi @hsztan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hsztan** [comment]
Thanks for your feedback, @peterrobert  😎

**Pull request merged**